src/python/foglamp/backup_restore/backup.py

```diff
@@ -0,0 +1,449 @@
+
+        self._config['port'] = int(self._config_from_manager['port']['value'])
+        self._config['database'] = self._config_from_manager['database']['value']
+        self._config['backup_dir'] = self._config_from_manager['backup_dir']['value']
+        self._config['retention'] = int(self._config_from_manager['retention']['value'])
+        self._config['timeout'] = int(self._config_from_manager['timeout']['value'])
+
+    def _retrieve_configuration_from_file(self):
+        """" Retrieves the configuration from the local file
+
+        Args:
+        Returns:
+        Raises:
+        Todo:
+        """
+
+        config_file = configparser.ConfigParser()
+        config_file.read(self._CONFIG_FILE)
+
+        self._config['host'] = config_file['DEFAULT']['host']
+        self._config['port'] = int(config_file['DEFAULT']['port'])
+        self._config['database'] = config_file['DEFAULT']['database']
+        self._config['backup_dir'] = config_file['DEFAULT']['backup_dir']
+        self._config['retention'] = int(config_file['DEFAULT']['retention'])
+        self._config['timeout'] = int(config_file['DEFAULT']['timeout'])
+
+    def _update_configuration_file(self):
+        """ Updates the configuration file with the values retrieved from tha manager.
+
+        Args:
+        Returns:
+        Raises:
+        Todo:
+        """
+
+        config_file = configparser.ConfigParser()
+
+        config_file['DEFAULT']['host'] = self._config['host']
+        config_file['DEFAULT']['port'] = str(self._config['port'])
+        config_file['DEFAULT']['database'] = self._config['database']
+        config_file['DEFAULT']['backup_dir'] = self._config['backup_dir']
+        config_file['DEFAULT']['retention'] = str(self._config['retention'])
+        config_file['DEFAULT']['timeout'] = str(self._config['timeout'])
+
+        with open(self._CONFIG_FILE, 'w') as file:
+            config_file.write(file)
+
+    def _retrieve_configuration(self):
+        """  Retrieves the configuration either from the manager or from a local file.
+        the local configuration file is used if the configuration manager is not available
+        and updated with the values retrieved from the manager when feasible.
+
+        Args:
+        Returns:
+        Raises:
+        Todo:
+        """
+
+        try:
+            self._retrieve_configuration_from_manager()
+
+        except Exception as _ex:
+            _message = self._MESSAGES_LIST["e000002"].format(_ex)
+            self._logger.warning(_message)
+
+            try:
+                self._retrieve_configuration_from_file()
+
+            except Exception as _ex:
+                _message = self._MESSAGES_LIST["e000003"].format(_ex)
+                self._logger.error(_message)
+
+                raise ConfigRetrievalError(ex)
+        else:
+            self._update_configuration_file()
+
+    def _purge_old_backups(self):
+        """  Deletes old backups in relation at the retention parameter
+
```

Comment:
Stefano and I discussed this and I asked Stefano to handling the purging in the backups and otherwise we have issues with space management and the need to track backup status.